Add Hardware settings doc page (#2518)
huggingface/hub-docs: docs/hub/hardware.md

# Hardware

> Tell the Hub which compute hardware you own (GPUs, CPUs, or Apple Silicon) and it will help you find models that run on your setup, show your setup on your profile, and let you compare it with the rest of the community.

Reviewer: ```suggestion
Tell the Hub which compute hardware you own (GPUs, CPUs, or Apple Silicon) and it will help you find models that run on your setup. You can also optionally share your hardware on your profile, and compare it with the rest of the community.
```

> Manage it any time from your [Hardware settings](https://huggingface.co/settings/hardware).

Reviewer: (Merged, but not yet deployed)

## Add your hardware

On your [Hardware settings](https://huggingface.co/settings/hardware) page, add each piece of hardware you own:

1. Pick a **type**: GPU, CPU, or Apple Silicon.
2. Choose the **provider** and **model** (for example, NVIDIA RTX 4090).
3. Set the **memory** (VRAM, RAM, or unified memory) and the **number of units** you have.

Add as many items as you like. If you own several, mark one as your **primary** hardware so it appears first on model pages.

> [!TIP]

> > Your hardware is private by default. Use the **Publicly Visible** toggle to show it on your profile.

Reviewer 2: euh are you sure? i thought it was the opposite

> > Your hardware is private by default. Use the **Publicly Visible** toggle to show it on your profile.

Reviewer: ah, did this use to be public by default?

Reviewer 2: still should be

Author: yes this was inaccurate it's public by default.

## See which models fit your hardware

> On model pages that offer [GGUF](./gguf) or MLX files, a **Hardware compatibility** panel estimates whether each quantization will run on your saved hardware, so you can choose a size that fits before downloading. Pair it with [Local Apps](./local-apps) to go from "will it run?" to running it in a couple of clicks.

Reviewer: ```suggestion
On model pages that offer [GGUF](./gguf) or MLX files, a **Hardware compatibility** panel estimates whether each quantization will run on your saved hardware, so you can choose a size that fits before downloading. Pair it with [Local Apps](./local-apps) to get up and running in a couple of clicks.
```

## Share and compare

Once your hardware is public:

> - A **TFLOPS** badge appears on your profile, summarizing your total compute power.

Reviewer: ```suggestion
- A **TFLOPS** badge appears on your profile, summarizing your estimated total compute power.
```

- You can browse [what the community is running](https://huggingface.co/hardware) and see how your setup compares across GPUs, CPUs, and Apple Silicon.

## Next steps

- [Use AI Models Locally](./local-apps) — run models with your favorite local app.

> - [Local Agents with llama.cpp](./agents-local) — build a coding agent on your own hardware.

Reviewer: 🔥
